.travis.yml

```diff
@@ -1,28 +1,37 @@
-# See https://github.com/silverstripe/silverstripe-travis-support for setup details
-
 language: php
 
-dist: precise
-
-sudo: false
-
-php:
-  - 5.6
-  - 7.0
-  - 7.1
-
 env:
-  - DB=MYSQL CORE_RELEASE=4
+  global:
+    - COMPOSER_ROOT_VERSION="4.0.x-dev"
+
+matrix:
+  include:
+    - php: 5.6
+      env: DB=MYSQL PHPCS_TEST=1 PHPUNIT_TEST=1
+    - php: 7.0
+      env: DB=PGSQL PHPUNIT_TEST=1
+    - php: 7.1
+      env: DB=MYSQL PHPUNIT_COVERAGE_TEST=1
 
 before_install:
   - echo "extension=ldap.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
 
 before_script:
-  - composer self-update || true
   - phpenv rehash
-  - git clone git://github.com/silverstripe/silverstripe-travis-support.git ~/travis-support
-  - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
-  - cd ~/builds/ss
+  - phpenv config-rm xdebug.ini
+
+  - composer install --prefer-dist
+  - composer require --prefer-dist --no-update silverstripe/framework:4.0.x-dev
```

Comment:
We should use `silverstripe/recipe-core 1.0.x-dev` here which will centralise some of these dependencies